cuda/base/cublas_bindings.hpp

```diff
@@ -90,7 +90,8 @@ struct is_supported<std::complex<double>> : std::true_type {};
             CublasName(handle, transa, transb, m, n, k, as_culibs_type(alpha), \
                        as_culibs_type(a), lda, as_culibs_type(b), ldb,         \
                        as_culibs_type(beta), as_culibs_type(c), ldc));         \
-    }
+    }                                                                          \
+    void __gko_macro_terminator__()
```

Comment:
Any reason you are not using:
```suggestion
    static_assert(true, "")
```
This would not add a dummy function to every class you use the macro while still needing the semicolon at the end.

Also, all names with double underscore are reserved according to the [C++ standard](http://www.lcdf.org/c%2B%2B/clause17.html#s17.4.3.1), so the name has to change:
> Certain sets of names and function signatures are always reserved to the implementation: 
> * Each name that contains a double underscore (_ _) or begins with an underscore followed by an upper case letter (2.11) is reserved to the implementation for any use.

For a more detailed overview, see stackoverflow answer: https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier

Reply:
Yeah, that does look better. I thought the `__gko_macro_terminator__()` shows its intent better. Using the `static_assert` seemed like the statement would be out of place. But maybe static assert is applicable to all situations and will not have overload problems as for the function. Thanks!

Reply:
In that case, what about:
```c++
static_assert(true, "This assert is only used to enforce a semicolon after the macro call.")
```
or similar. You can write in the message whatever you want, therefore, it is a nice place for a comment.

Reply:
That works. Thanks!